.github/workflows/check_bom.yml

```diff
@@ -0,0 +1,49 @@
+name: BOM Check
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  bom-check:
+    if: ${{ github.event_name != 'pull_request' || github.repository != github.event.pull_request.head.repo.full_name }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: UTF-8 BOM
+        run: |          
+          if grep -lrE --include="*.c" $'\xEF\xBB\xBF' .; then exit 1; else exit 0; fi
+          if grep -lrE --include="*.h" $'\xEF\xBB\xBF' .; then exit 1; else exit 0; fi
```

Comment:
I tested and the CI only catches `.c` files. If all the C files are okay, it runs `exit 0`, skipping the rest of the steps.

Reply:
Ohhhh.... I know what is happening. The CI is interpreting each exit call as a literal exit from the CI and not an exit from the script. that is why the next lines don't run.


Just need to remove the `else exit 0;` from each line.

Reply:
OK this issue should be fixed now..